### PR TITLE
Significantly improve performance on par with virtual dom diffing

### PR DESCRIPTION
**Pull request by snewcomer**
Benchmarks show [morphdom](https://github.com/patrick-steele-idem/morphdom) is 2x slower on large diffs with virtual DOM.  Small diffs (probably 80% of the use cases with live_view) are ~same with real DOM vs. virtual DOM.

This change puts morphdom on par with virtual DOM on large diffs and makes some small diff cases faster as well.

Leaving this open for now to investigate as onBeforeElUpdated and onElUpdated will not be called for the fromEl AND all of its children if this change makes it in.  My initial inclination is this isn't a problem

### CODE REVIEW

File: assets/js/phoenix_live_view.js [hunk 1]
```diff
@@ -533,6 +533,10 @@ let DOM = {
         }
       },
       onBeforeElUpdated: function(fromEl, toEl) {
+        if (fromEl.isEqualNode(toEl)) {
+           return false // Skip this entire sub-tree
+        }
```
Comment (snewcomer) [inline assets/js/phoenix_live_view.js]: I don't think this is a problem b/c of what the spec says about `isEqualNode` and deep comparisons including children and the index they are at - https://dom.spec.whatwg.org/#concept-node-equals

**Comment by raygesualdo** [comment]
Any before/after benchmarks?

**Comment by snewcomer** [comment]
@raygesualdo Just the benchmarks in morphdom when adding this to the source code.  I am not necessarily hesitant to add to the library; however, being unfamiliar with the whole history of the project and use cases, I have held off.  For live_view though, this is totally doable and effectively avoid traversing down the tree when the DOM can already tell us everything is the same.

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**